spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -685,7 +685,7 @@ This module uses the C preprocessor to select a target architecture.
 > maybeReturnSc ntfnPtr tcbPtr = do
 >     nscOpt <- liftM ntfnSc $ getNotification ntfnPtr
 >     tscOpt <- threadGet tcbSchedContext tcbPtr
->     when (nscOpt == tscOpt) $ do
+>     when (nscOpt == tscOpt && nscOpt /= Nothing) $ do
 >         threadSet (\tcb -> tcb { tcbSchedContext = Nothing }) tcbPtr
 >         scPtr <- return $ fromJust nscOpt
 >         sc <- getSchedContext scPtr
```

Comment:
It looks like the Haskell and abstract versions of this function are different from the C. The C code doesn’t seem to do rescheduling. Do you mind checking this out, Mitch?

Reply:
Will do.

Reply:
I hope that our email discussions have resolved this (This change is necessary but was lost in some changes on the systems side. It is now recorded in commits against cmillar/mcs-verified or cmillar/mcs-target)

Reply:
It's currently tied up in `sporadic-fixes`, I should be able to clean that branch up and make some actual distinct PRs there next week